tests/test_other.py

```diff
@@ -10552,3 +10552,17 @@ def test_bitcode_input(self):
     self.assertTrue(shared.Building.is_bitcode('main.bc'))
     run_process([PYTHON, EMCC, '-c', '-o', 'main.o', 'main.bc'])
     self.assertTrue(shared.Building.is_wasm('main.o'))
+
+  def test_nostdlib(self):
+    # First ensure all the system libs are built
+    run_process([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c')])
+
+    self.assertContained('undefined symbol:', self.expect_fail([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nostdlib']))
+    self.assertContained('undefined symbol:', self.expect_fail([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nodefaultlibs']))
+
+    # Buil again but with explit system libraries
+    libs = ['-lc', '-lcompiler_rt']
+    if self.is_wasm_backend():
+      libs.append('-lc_rt_wasm')
+    run_process([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nostdlib'] + libs)
+    run_process([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nodefaultlibs'] + libs)
```

Comment:
can have one big loop on the outside for the two `-no*` flags, to avoid 2 sets of duplicate lines?

Reply:
I have an aversion to loops like that in test code since it makes failure messages harder to interpret.     if it work many duplicate lines I might agree with you, and make parenthesized test.  In this case I'm ok with the duplication level.

Reply:
Sounds good.